### PR TITLE
Info.plist generation from the bundler configuration file property

### PR DESCRIPTION
**Pull request by sam-kamerer**
I needed to have different **Info.plist** files for several programs. And I decided to put the configuration for **Info.plist** into the bundler configuration file (**bundler.json**).

### CODE REVIEW

File: bundler.go [hunk 1]
```diff
@@ -607,8 +613,18 @@ func (b *Bundler) finishDarwin(environmentPath, binaryPath string) (err error) {
 		return
 	}
 
-	// Move binary
 	var macOSBinaryPath = filepath.Join(macOSPath, b.appName)
+
+	var infoPlist *plister.InfoPlist
+	if b.infoPlist != nil {
+		infoPlist = plister.MapToInfoPlist(b.infoPlist)
+		binaryName, _ := infoPlist.Get("CFBundleExecutable").(string)
```
Comment (asticode) [inline bundler.go]: Could you still check the type assertion is successful ?

```go
if binaryName, ok := infoPlist.Get("CFBundleExecutable").(string); ok && len(binaryName) > 0 {
  macOSBinaryPath = filepath.Join(macOSPath, binaryName)
}
```

File: bundler.go [hunk 2]
```diff
@@ -637,8 +653,17 @@ func (b *Bundler) finishDarwin(environmentPath, binaryPath string) (err error) {
 			return
 		}
 
+		iconFileName := b.appName + filepath.Ext(b.pathIconDarwin)
+
+		if infoPlist != nil {
+			ifn, _ := infoPlist.Get("CFBundleIconFile").(string)
```
Comment (asticode) [inline bundler.go]: Same here

```go
if ifn, ok := infoPlist.Get("CFBundleIconFile").(string); ok && len(ifn) > 0 {
  iconFileName = ifn
}
```

File: bundler.go [hunk 3]
```diff
@@ -654,6 +679,11 @@ func (b *Bundler) finishDarwin(environmentPath, binaryPath string) (err error) {
 	// Add Info.plist file
 	var fp = filepath.Join(contentsPath, "Info.plist")
 	astilog.Debugf("Adding Info.plist to %s", fp)
+
+	if infoPlist != nil {
+		return plister.Generate(fp, infoPlist)
```
Comment (asticode) [inline bundler.go]: Can you wrap the error here ?

```go
if err = plister.Generate(fp, infoPlist); err != nil {
  err = errors.Wrap(err, "generating Info.plist failed")
  return
}
return
```

File: bundler.go (hunk 1 above)
Comment (sam-kamerer) [reply bundler.go]: In this case, we do not need to do redundant checks. It is enough for us to verify that the result of the casting is not equal to the empty string. Look at the example how it works https://goplay.space/#3LZLCttXr7U

One more thing. In the idiomatic Go, checking for empty strings should be done by comparing with an empty string, and not by comparing its length (https://dmitri.shuralyov.com/idiomatic-go#empty-string-check)

File: bundler.go (hunk 2 above)
Comment (sam-kamerer) [reply bundler.go]: Same as above

**Comment by asticode** [comment]
Cheers

**Pull request merged**